lib/config/fileconf.go

```diff
@@ -1188,6 +1190,20 @@ type AWSMatcher struct {
 	SSMDocument string `yaml:"ssm_command_document"`
 }
 
+// AzureMatcher matches Azure databases.
+type AzureMatcher struct {
+	// Subscriptions are Azure subscriptions to query for resources
+	Subscriptions []string `yaml:"subscriptions,omitempty"`
+	// ResourceGroups are Azure resource groups to query for resources.
+	ResourceGroups []string `yaml:"resource_groups,omitempty"`
+	// Types are azure database types to match: "mysql", "postgres"
+	Types []string `yaml:"types,omitempty"`
+	// Regions are Azure locations to match for databases.
+	Regions []string `yaml:"regions,omitempty"`
+	// Tags are Azure tags to match.
+	Tags map[string]apiutils.Strings `yaml:"tags,omitempty"`
+}
```

Comment:
Both `Subscriptions` and `ResourceGroups` support tags. I assume these `Tags` are only for resources?

Reply:
I couldn't find an RFD on this. Just curious if we have discussed whether we want to use tags to filter subscriptions and resource groups vs using explicit names.

Reply:
Tags are for resources, yes. I'll rename to "ResourceTags". We haven't discussed this anywhere before, and it actually didn't occur to me until you pointed it out. You can specify `*` for subscriptions/groups to get the behavior of only using resource tags for filtering. Or you can be more specific about which subscriptions/groups you want to query. This already enables more specific queries than we support for AWS.

Looking at the SDK, it would require quite a bit more code to watch for subscriptions and resource groups by tag, and then spin up watchers for resources within those subscriptions and groups. Not only that, but there is no API query for finding the subscriptions and resource groups by tag - you would need to query for all of them and then filter the results by tag.

Reply:
Sounds good. We can keep it simple for now.